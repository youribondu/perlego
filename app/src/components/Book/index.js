import React from "react";
import "./index.scss";
import { LazyLoadImage } from "react-lazy-load-image-component";

const Book = ({ title = "", authors = [], cover = "" }) => {
  return (
    <div className="p-book">
      <LazyLoadImage alt={title} effect="blur" src={cover} />
      <div className="p-book__info">
        <h3 data-testid="book-title">{title}</h3>
        <p data-testid="book-author">{authors[0]}</p>
      </div>
    </div>
  );
};

export default Book;
