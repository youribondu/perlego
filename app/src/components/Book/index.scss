@import "../../styles/index.scss";
@import "react-lazy-load-image-component/src/effects/blur.css";

.p-book {
  display: flex;
  flex-direction: column;
  height: 100%;

  img {
    width: 100%;
    border-radius: $border-radius;
  }

  &__info {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    flex-grow: 1;

    p {
      color: $color-grey;
    }
  }
}
