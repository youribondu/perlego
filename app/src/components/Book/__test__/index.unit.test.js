import React from "react";
import ReactDom from "react-dom";
import Book from "../index";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer";

describe("Book Component", () => {
  afterEach(cleanup);

  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Book />, div);
  });

  it("Renders book correctly", () => {
    const title = "title";
    const authors = ["author1", "author2"];
    const cover = "cover";
    const { getByTestId } = render(
      <Book title={title} authors={authors} cover={cover} />
    );
    expect(getByTestId("book-title")).toHaveTextContent(title);
    expect(getByTestId("book-author")).toHaveTextContent(authors[0]);
  });

  it("Matches snapshot empty", () => {
    const book = create(<Book />).toJSON();
    expect(book).toMatchSnapshot();
  });

  it("Matches snapshot with props", () => {
    const title = "title";
    const authors = ["author1", "author2"];
    const cover = "cover";
    const book = create(
      <Book title={title} authors={authors} cover={cover} />
    ).toJSON();
    expect(book).toMatchSnapshot();
  });
});
