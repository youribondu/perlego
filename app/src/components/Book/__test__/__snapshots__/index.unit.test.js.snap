// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Component Matches snapshot empty 1`] = `
<div
  className="p-book"
>
  <span
    className=" lazy-load-image-background blur"
    style={
      Object {
        "backgroundImage": "",
        "backgroundSize": "",
        "color": "transparent",
        "display": "inline-block",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
  </span>
  <div
    className="p-book__info"
  >
    <h3
      data-testid="book-title"
    >
      
    </h3>
    <p
      data-testid="book-author"
    />
  </div>
</div>
`;

exports[`Book Component Matches snapshot with props 1`] = `
<div
  className="p-book"
>
  <span
    className=" lazy-load-image-background blur"
    style={
      Object {
        "backgroundImage": "",
        "backgroundSize": "",
        "color": "transparent",
        "display": "inline-block",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
  </span>
  <div
    className="p-book__info"
  >
    <h3
      data-testid="book-title"
    >
      title
    </h3>
    <p
      data-testid="book-author"
    >
      author1
    </p>
  </div>
</div>
`;
